rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/CatalogRequestBuilder.java

```diff
@@ -27,93 +27,38 @@
 import com.google.gwt.http.client.RequestException;
-import com.google.gwt.http.client.Response;
-import com.google.gwt.json.client.JSONArray;
-import com.google.gwt.json.client.JSONObject;
-import com.google.gwt.json.client.JSONParser;
 
 
 public class CatalogRequestBuilder {
 
     public static final String SESSION_ID_PARAMETER_NAME = "sessionId";
 
-    private static final String NAME_KEY = "name";
-
-    private ImportFromCatalogPanel importFromCatalogPanel;
-
     private String catalogUrl;
 
-    public CatalogRequestBuilder(ImportFromCatalogPanel importFromCatalogPanel) {
-        this.importFromCatalogPanel = importFromCatalogPanel;
-        this.catalogUrl = new CatalogUrlBuilder().getCatalogUrl();
-    }
-
-    public void requestNodeSourcesFromAllBuckets() {
-        String listBucketsUrl = this.catalogUrl + "/buckets";
-        sendRequestToCatalog(listBucketsUrl, getBucketsRequestCallBack());
-    }
-
-    private RequestCallback getBucketsRequestCallBack() {
-        return new RequestCallback() {
-            @Override
-            public void onResponseReceived(Request request, Response response) {
-                JSONArray bucketsArray = JSONParser.parseStrict(response.getText()).isArray();
-                for (int i = 0; i < bucketsArray.size(); i++) {
-                    JSONObject bucketObject = bucketsArray.get(i).isObject();
-                    String bucketName = bucketObject.get(NAME_KEY).isString().stringValue();
-                    requestNodeSourcesForBucket(bucketName);
-                }
-                CatalogRequestBuilder.this.importFromCatalogPanel.enableNodeSourceListBox();
-            }
+    private ImportFromCatalogPanelFeeder importFromCatalogPanelFeeder;
 
-            @Override
-            public void onError(Request request, Throwable t) {
-                CatalogRequestBuilder.this.importFromCatalogPanel.setNodeSourceWindowLabelWithError("List buckets from catalog failed",
-                                                                                                    t);
-            }
-        };
-    }
-
-    private void requestNodeSourcesForBucket(String bucketName) {
-        String nodeSourcesListUrl = this.catalogUrl + "/buckets/" + bucketName + "/resources?kind=nodesource";
-        sendRequestToCatalog(nodeSourcesListUrl, getNodeSourcesRequestCallback(bucketName));
+    public CatalogRequestBuilder(ImportFromCatalogPanelFeeder importFromCatalogPanelFeeder) {
+        this.catalogUrl = new CatalogUrlBuilder().getCatalogUrl();
+        this.importFromCatalogPanelFeeder = importFromCatalogPanelFeeder;
     }
 
-    private void sendRequestToCatalog(String url, RequestCallback callback) {
-        RequestBuilder request = new RequestBuilder(RequestBuilder.GET, url);
+    public void sendRequestToCatalog(String endpoint, RequestCallback callback) {
+        String endpointToAddToCatalogUrl = endpoint;
+        if (endpoint.startsWith("/")) {
```

Comment:
Are there indeed two possible use cases: with and without slash?

Reply:
Not in the current codebase. I removed it.